python/main.py

```diff
@@ -0,0 +1,583 @@
+        pvd = ParamVariableDeclaration()
+        self.add_loc_info(node, pvd)
+        pvd.setName(node.arg)
+        # TODO handle annotation
+        return pvd
+
+    def visit_Return(self, node):
+        debug_print(ast.dump(node))
+        r = ReturnStatement()
+        self.add_loc_info(node, r)
+        r.setReturnValue(self.visit(node.value))
+        return r
+
+    def visit_Yield(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_YieldFrom(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_Global(self, node):
+        debug_print(ast.dump(node))
+        if len(node.names) != 1:
+            raise NotImplementedError
+        ref = DeclaredReferenceExpression()
+        self.add_loc_info(node, ref)
+        ref.setName(node.names[0])
+        return ref
+
+    def visit_Nonlocal(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    def visit_ClassDef(self, node):
+        debug_print(ast.dump(node))
+        rec = RecordDeclaration()
+        self.add_loc_info(node, rec)
+        # name
+        rec.setName(node.name)
+        # bases
+        if len(node.bases) == 0:
+            pass
+        elif len(node.bases) == 1:
+            b = node.bases[0]
+            if isinstance(b, ast.Attribute):
+                tname = "%s.%s" % (b.value.id, b.attr)
+                t = TypeParser.createFrom(tname, True)
+            else:
+                raise NotImplementedError
+        else:
+            raise NotImplementedError
+        rec.setSuperClasses([t])
+        if len(node.keywords) != 0:
+            raise NotImplementedError
+        #if node.starargs is not None:
+        #    raise NotImplementedError
+        #if node.kwargs is not None:
+        #    raise NotImplementedError
+        # body
+        for b in node.body:
+            if isinstance(b, ast.FunctionDef):
+                fd = self.visit_FunctionDef(b, returnMethod=True)
+                rec.addMethod(fd)
+            else:
+                raise NotImplementedError
+
+        if len(node.decorator_list) != 0:
+            raise NotImplementedError
+
+
+        debug_print(rec)
+
+        return rec
+
+    ### ASYNC AND AWAIT ###
+    def visit_AsyncFunctionDef(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_Await(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_AsyncFor(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_AsyncWith(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### MISC ###
+    def visit_Module(self, node):
+        debug_print(ast.dump(node))
+        '''The AST is wrapped in a 'Module' with 'body' list. Visit the body.'''
+        for n in node.body:
+            self.tud.addDeclaration(self.visit(n))
```

Comment:
Same as above. the global layer is a special case. In this case, enterScope should not be used, but instead ScopeManager::resetGlobal(tu). This will then tie the current global scope to the current translation unit.

BTW: `tu` is mostly used as abbreviation for a translation unit in other langues.